cranelift/filetests/src/test_zkasm.rs

```diff
@@ -0,0 +1,187 @@
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use cranelift_codegen::entity::EntityRef;
+    use cranelift_codegen::ir::function::FunctionParameters;
+    use cranelift_codegen::ir::ExternalName;
+    use cranelift_codegen::isa::zkasm;
+    use cranelift_codegen::{settings, FinalizedMachReloc, FinalizedRelocTarget};
+    use cranelift_wasm::{translate_module, DummyEnvironment};
+
+    fn generate_preamble(start_func_index: usize) -> Vec<String> {
+        let mut program: Vec<String> = Vec::new();
+        program.push("start:".to_string());
+        program.push("  zkPC + 2 => RR".to_string());
+        program.push(format!("  :JMP(function_{})", start_func_index));
+        program.push("  :JMP(finalizeExecution)".to_string());
+        program
+    }
+
+    fn generate_postamble() -> Vec<String> {
+        let mut program: Vec<String> = Vec::new();
+        program.push("finalizeExecution:".to_string());
```

Comment:
Out of curiosity, what is the point of this? From the presentation we've seen last week, IIRC Polygon folks use this to reset the registers back to their initial state or some such, but we seem to never do this really.